### ISSUE TITLE
Gazelle requires WORKSPACE when using update-repos in a bzlmod only repo

### ISSUE THREAD

**Issue by Megakuul** [bazelbuild/bazel-gazelle#1726]
### What version of gazelle are you using?

0.35.0

### What version of rules_go are you using?

0.44.2

### What version of Bazel are you using?

7.0.1

### Does this issue reproduce with the latest releases of all the above?

Yes.

### What operating system and processor architecture are you using?

Linux /  x86_64

### What did you do?

I ran gazelle-update-repos in a project that does not use a WORKSPACE file but instead the MODULES.bazel file (bzlmod).

Update-repos script:

```bazel
gazelle(
    name = "gazelle-update-repos",
    args = [
        "-from_file=go.mod",
        "-to_macro=go_deps.bzl%go_dependencies",
        "-bzlmod",
        "-prune",
    ],
    command = "update-repos",
)
```

### What did you expect to see?

I expect it to just generate the `go_deps.bzl` file when bzlmod is enabled and ignore the WORKSPACE file.

If I understand this right, the pr #1589 already made it not update the WORKSPACE if bzlmod is enabled. For Bzlmod only projects, I assume that the issue is right here: https://github.com/bazelbuild/bazel-gazelle/blob/0205c200e7bc61270e7b1b2f081df929289e6aa1/cmd/gazelle/update-repos.go#L109-L118

### What did you see instead?

I got an error that it was unable to find the WORKSPACE file: 
`gazelle: loading WORKSPACE file: open .../WORKSPACE: no such file or directory`.

**Comment by Megakuul** [comment]
I've created a proposal PR that omits reading the WORKSPACE repositories if repo is bzlmod only.

**Comment by fmeum** [comment]
I understand the issue, but if you aren't using WORKSPACE anymore, why do you even need `update-repos` for Go deps?

**Comment by Megakuul** [comment]
The situation is the following: I used `update-repos` to manage Go dependencies from `go.mod` files (so that I only need to manage the versions of dependencies in one place). Now, after fully migrating everything from `WORKSPACE` to `MODULE.bazel`, I still want to manage the Go dependencies in one place. The most straightforward way that I found was to just generate the `deps.bzl` like usual and then just include it with `use_extension` and `use_repo`.

But maybe I'm on the completely wrong path...
